### ISSUE TITLE
GPM support in Twin Term 

### ISSUE THREAD

**Issue by RealSuperRyn** [cosmos72/twin#89]
It would be amazing for productivity in Twin if the terminals inside it had mouse support. You could use Midnight Commander as if it were a GUI file manager in a graphical environment, without having to remember tons of keybinds.

If this is already possible in a TTY, please let me know.

**Comment by Mirppc** [comment]
it is already possible.  gpm is one of the dependencies.  most distros disabled the sevice.

**Comment by RealSuperRyn** [comment]
> it is already possible. gpm is one of the dependencies. most distros disabled the sevice.

I already have GPM enabled on my system. The issue is that my mouse inputs aren't detected by the applications in the terminals. If i open Midnight commander in my TTY, I can click various things with my mouse and it'll select them, but if I open Twin, open an instance of Twin Term, and open Midnight commander in there, that functionality is lost.

**Comment by Mirppc** [comment]
This was an issue with YAST on my Opensuse machines where if i opened YAST via a TTY/SSH session running twin i would lose GPM.  Right now i seem to be having more issues building the latest version so i will have to get back to you after i get a build going and do some testing.

**Comment by cosmos72** [comment]
GPM is needed for mouse support when running twin on Linux console - it's not needed in other cases.

The terminals inside twin set `$TERM=linux`, which usually does not *directly* support mouse: when running on an actual Linux console, programs need to manually connect to the GPM daemon to receive mouse events.
Twin terminals extend the functionality of `$TERM=linux`, also providing xterm-style mouse events, if a program enables them by sending the appropriate escape sequences.

But lots of mouse-aware programs, including Midnight Commander, check the environment variable `$TERM` to decide whether the terminal supports mouse or not: for `$TERM=linux` they do not even try to enable xterm-style mouse events.

You can try to cheat a little and manually set `$TERM=xterm` before running Midnight Commander - let me know if it helps.

**Comment by RealSuperRyn** [comment]
> You can try to cheat a little and manually run `TERM=xterm; export TERM` before running Midnight Commander - let me know if it helps.
While it did "work", it ended up causing garbled output.
![whar](https://github.com/cosmos72/twin/assets/112213321/871a21ca-20fb-4543-9a4c-16fc8f94302a)

**Comment by Lassebq** [comment]
Can confirm. Only applications which use xterm way of handling mouse inputs seem to work. vim and midnight commander (the only ones I know which use GPM directly) are broken and ignore mouse clicks in twin.

**Comment by cosmos72** [comment]
Yes, because when vim and midnight commander see the environment variable `TERM=linux` (twin terminals emulate a Linux console) they assume to be running **directly** inside an actual Linux console, thus they try to use GPM for mouse support.

As I wrote above, a possible workaround (which may not work, as `xterm` and `linux` terminals are not fully compatible), is to slightly cheat and set `TERM=xterm` before launching vim, midnight commander or other GPM-aware programs.

The better solution would be to search in vim and midnight commander's documentation for how to enable xterm-compatibile mouse support also when `TERM` is different from `xterm`

**Comment by Lassebq** [comment]
Setting `TERM` does the trick, sorry for confusion

**Comment by cosmos72** [comment]
Version 1.0.0 is now released, and sets `$TERM=xterm-256color` that should help programs to detect and use xterm-style mouse support.

Programs running inside twin cannot use GPM - and should not even try - because GPM can only be used by programs running **directly** inside a real Linux console
